File: csharp_body_deref/Scripts/Settings/SkyboxCustomSkinSet.cs
// ==================== AoTTG2 cross-reference ====================
// Type: Settings.SkyboxCustomSkinSet
// Update status: unchanged since the previous dump
// Real method bodies (native C decompile): source/cpp/Settings/SkyboxCustomSkinSet.c
// Prior real C# source (older reference): Assets/Scripts/Settings/CustomSkins/SkyboxCustomSkinSet.cs
// Bodies below are empty IL2CPP metadata stubs; the logic lives in the native C.
// ===============================================================
using Il2CppDummyDll;

namespace Settings;

[Token(Token = "0x200006E")]
internal class SkyboxCustomSkinSet : BaseSetSetting
{
	[Token(Token = "0x400022F")]
	[FieldOffset(Offset = "0x38")]
	public StringSetting Front;

	[Token(Token = "0x4000230")]
	[FieldOffset(Offset = "0x40")]
	public StringSetting Back;

	[Token(Token = "0x4000231")]
	[FieldOffset(Offset = "0x48")]
	public StringSetting Left;

	[Token(Token = "0x4000232")]
	[FieldOffset(Offset = "0x50")]
	public StringSetting Right;

	[Token(Token = "0x4000233")]
	[FieldOffset(Offset = "0x58")]
	public StringSetting Up;

	[Token(Token = "0x4000234")]
	[FieldOffset(Offset = "0x60")]
	public StringSetting Down;

	[Token(Token = "0x6000315")]
	[Address(RVA = "0x3F86170", Offset = "0x3F86170", VA = "0x3F86170")]
	public SkyboxCustomSkinSet()
	{
	}
}

File: cpp/Settings/SkyboxCustomSkinSet.c
// Type: Settings.SkyboxCustomSkinSet
// Ghidra (System V/gcc) decompilation, IL2CPP structs + signatures applied.
// ---- AoTTG2 cross-reference ----
// Update status: unchanged since the previous dump
// C# structure: source/csharp/Scripts/Settings/SkyboxCustomSkinSet.cs
// Prior real C# source (older reference): Assets/Scripts/Settings/CustomSkins/SkyboxCustomSkinSet.cs
// --------------------------------

// Settings.SkyboxCustomSkinSet$$.ctor
// il2cpp: void Settings_SkyboxCustomSkinSet___ctor (Settings_SkyboxCustomSkinSet_o* __this, const MethodInfo* method);
// 0x4086170

void Settings_SkyboxCustomSkinSet___ctor(Settings_SkyboxCustomSkinSet_o *__this,MethodInfo *method)

{
  System_String_o *pSVar1;
  Settings_StringSetting_o *pSVar2;
  
  if (g_data_057ac3da == '\0') {
    il2cpp_runtime_helper_023445d0(&TypeInfo_StringSetting);
    g_data_057ac3da = '\x01';
  }
  pSVar1 = (System_String_o *)**(undefined8 **)(g_data_057b9c00 + 0xb8);
  pSVar2 = (Settings_StringSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_StringSetting);
  Settings_StringSetting___ctor_40f74f0(pSVar2,pSVar1,200,(MethodInfo *)0x0);
  (__this->fields).Front = pSVar2;
  il2cpp_runtime_helper_022b4080(&(__this->fields).Front,pSVar2);
  pSVar1 = (System_String_o *)**(undefined8 **)(g_data_057b9c00 + 0xb8);
  pSVar2 = (Settings_StringSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_StringSetting);
  Settings_StringSetting___ctor_40f74f0(pSVar2,pSVar1,200,(MethodInfo *)0x0);
  (__this->fields).Back = pSVar2;
  il2cpp_runtime_helper_022b4080(&(__this->fields).Back,pSVar2);
  pSVar1 = (System_String_o *)**(undefined8 **)(g_data_057b9c00 + 0xb8);
  pSVar2 = (Settings_StringSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_StringSetting);
  Settings_StringSetting___ctor_40f74f0(pSVar2,pSVar1,200,(MethodInfo *)0x0);
  (__this->fields).Left = pSVar2;
  il2cpp_runtime_helper_022b4080(&(__this->fields).Left,pSVar2);
  pSVar1 = (System_String_o *)**(undefined8 **)(g_data_057b9c00 + 0xb8);
  pSVar2 = (Settings_StringSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_StringSetting);
  Settings_StringSetting___ctor_40f74f0(pSVar2,pSVar1,200,(MethodInfo *)0x0);
  (__this->fields).Right = pSVar2;
  il2cpp_runtime_helper_022b4080(&(__this->fields).Right,pSVar2);
  pSVar1 = (System_String_o *)**(undefined8 **)(g_data_057b9c00 + 0xb8);
  pSVar2 = (Settings_StringSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_StringSetting);
  Settings_StringSetting___ctor_40f74f0(pSVar2,pSVar1,200,(MethodInfo *)0x0);
  (__this->fields).Up = pSVar2;
  il2cpp_runtime_helper_022b4080(&(__this->fields).Up,pSVar2);
  pSVar1 = (System_String_o *)**(undefined8 **)(g_data_057b9c00 + 0xb8);
  pSVar2 = (Settings_StringSetting_o *)il2cpp_runtime_helper_023052d0(TypeInfo_StringSetting);
  Settings_StringSetting___ctor_40f74f0(pSVar2,pSVar1,200,(MethodInfo *)0x0);
  (__this->fields).Down = pSVar2;
  il2cpp_runtime_helper_022b4080(&(__this->fields).Down);
  Settings_BaseSetSetting___ctor((Settings_BaseSetSetting_o *)__this,(MethodInfo *)0x0);
  return;
}


